contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -928,6 +929,177 @@ checksum(PG_FUNCTION_ARGS)
+	Oid 			schema_oid;
+	Oid 			result = InvalidOid;
+	int 			i;
+	char 			*input = lowerstr(text_to_cstring(PG_GETARG_TEXT_P(0)));
+	char 			*object_type = text_to_cstring(PG_GETARG_TEXT_P(1));
+	List 			*splited_object_name;
+
+	/* strip trailing whitespace */
+	i = strlen(input);
+	while (i > 0 && isspace((unsigned char) input[i - 1]))
+		input[--i] = '\0';
+	
+	/* resolve the three part name */
+	splited_object_name = split_object_name(input);
+	switch (list_length(splited_object_name))
+	{
+		case 1:
+			object_name = (char *) linitial(splited_object_name);
+			break;
+		case 2:
+			schema_name = (char *) linitial(splited_object_name);
+			object_name = (char *) lsecond(splited_object_name);
+			break;
+		case 3:
+			db_name = (char *) linitial(splited_object_name);
+			schema_name = (char *) lsecond(splited_object_name);
+			object_name = (char *) lthird(splited_object_name);
+			break;
+		default:
+			list_free(splited_object_name);
+			PG_RETURN_NULL();
+	}
+	pfree(input);
+	list_free(splited_object_name);
+
+	/* truncate identifiers if needed */
+	truncate_tsql_identifier(db_name);
+	truncate_tsql_identifier(schema_name);
+	truncate_tsql_identifier(object_name);
+
+	if (!strcmp(db_name, ""))
+		db_name = get_cur_db_name();
+
+	/* check if looking in current database, if not then return null value */
+	if (strcmp(db_name, get_cur_db_name()) && strcmp(db_name, "tempdb"))
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_DATABASE),
+				 errmsg("Can only do lookup in current database.")));
+
+	if (!strcmp(schema_name, ""))
+	{	
+		/* find the default schema for current user */
+		const char *user = get_user_for_database(db_name);
+		schema_name = get_authid_user_ext_schema_name((const char *) db_name, user);
+	}
+
+	/* get physical schema name from logical schema name */
+	physical_schema_name = get_physical_schema_name(db_name, schema_name);
+
+	/* get schema oid from physical schema name */
+	schema_oid = LookupExplicitNamespace(physical_schema_name, true);
+
+	/* free unnecessary pointers */
+	pfree(db_name);
+	pfree(schema_name);
+	pfree(physical_schema_name);
+
+	if (!OidIsValid(schema_oid))
+	{
+		pfree(object_name);
+		pfree(object_type);
+		PG_RETURN_NULL();
+	}
+	
+	/* check if looking for temp object */
+	is_temp_object = (object_name[0] == '#'? true: false);
+
+	if (strcmp(object_type, "")) /* object_type is not empty */
+	{
+		if (!strcmp(object_type, "S") || !strcmp(object_type, "U") || !strcmp(object_type, "V") ||
+			!strcmp(object_type, "IT") || !strcmp(object_type, "ET") || !strcmp(object_type, "SO"))
+		{
+			if (is_temp_object)
+			{
+				/* search in list of ENRs registered in the current query environment by name */
+				result = tsql_get_temp_object_oid(object_name);
+			}
+			else
+			{
+				/* search in pg_class by name and schema oid */
+				result = get_relname_relid((const char *) object_name, schema_oid); 
+			}
+		}
+		else if (!strcmp(object_type, "C") || !strcmp(object_type, "D") || !strcmp(object_type, "F") ||
```

Comment:
Shouldnt we return NULL for other object types if their names start with '#'?
is_temp_object would be true, but we are ignoring it in further scenarios